test/object_store/test_backup.py

```diff
@@ -176,3 +178,7 @@ def list_sstables():
     res = cql.execute(f"SELECT * FROM {ks}.{cf};")
     rows = { x.name: x.value for x in res }
     assert rows == orig_rows, "Unexpected table contents after restore"
+
+    print(f'Check that backup files are still there') # regression test for #20938
+    post_objects = set([ o.key for o in get_s3_resource(s3_server).Bucket(s3_server.bucket_name).objects.all() ])
+    assert objects == post_objects
```

Comment:
weird ... 

seems the sstables are still being removed

```
        post_objects = set([ o.key for o in get_s3_resource(s3_server).Bucket(s3_server.bucket_name).objects.all() ])
>       assert objects == post_objects
E       AssertionError: assert {'test_cf/bac...ndex.db', ...} == {'test_cf/bac...ndex.db', ...}
E         Extra items in the right set:
E         'test_cf/backup_1727945696871_fyqcg/me-3gk2_0ork_5lh342pdpw36obm0p3-big-Statistics.db'
E         'test_cf/backup_1727945696868_lnqeb/me-3gk2_0ork_5jbxc2l9veu6lp0uct-big-Scylla.db'
E         'test_cf/backup_1727945696868_lnqeb/me-3gk2_0ork_5jbxc2l9veu6lp0uct-big-Summary.db'
E         'test_cf/backup_1727945696865_qlcra/me-3gk2_0ork_5iwhs1za70t5n6eza2-big-Digest.crc32'
E         'test_cf/backup_1727945696865_qlcra/me-3gk2_0ork_5iwhs1za70t5n6eza2-big-Filter.db'
E         'test_cf/backup_1727945696871_fyqcg/me-3gk2_0ork_5lh342pdpw36obm0p3-big-Index.db'...
E         
E         ...Full output truncated (55 lines hidden), use '-vv' to show
```

see https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12155/testReport/junit/(root)/test_backup/test_simple_backup_and_restore_dev_1/

Reply:
Nope. it's extra objects on the _right_ side -- i.e. -- some objects appeared there after restore. I guess it's shared bucket problem :( Need some more elaborated check

Reply:
heh, the nuke-the-world button. how about using the fixture at https://github.com/scylladb/scylladb/pull/20863/files#diff-f9e108684826efb9b554dfcf2b4a0eed389f6f240986675732d090c35c0b77caR1263 ?